### ISSUE TITLE
AddBlockRequestProto Not Set FileId Cause FileWriter.Write Error

### ISSUE THREAD

**Issue by getumen** [colinmarc/hdfs#299]
error message
```
addBlock call failed with ERROR_APPLICATION (org.apache.hadoop.hdfs.server.namenode.LeaseExpiredException)
```
In Hadoop 3.x, when I uploaded a file larger than block size, LeaseExpiredException occured.
That is because AddBlockRequestProto doesn't have FileID in `addBlock` request. https://github.com/colinmarc/hdfs/blob/2f114063eda00d5847ff83c07afb9ad04e90f7b4/file_writer.go#L266-L270

I don't have, myself, full confidence that this error is reproducible in other hadoop environment.

**Comment by getumen** [comment]
This issue (https://github.com/colinmarc/hdfs/issues/298) may be same reason.

**Comment by 982945902** [comment]
you can look at https://github.com/colinmarc/hdfs/pull/301

**Comment by colinmarc** [comment]
Fixed in #301.
